File: Array/Basics/ReverseArray.go
package main

import "fmt"

// reverseArray reverses the given array in place.
func reverseArray(arr []int) []int {
    
	start := 0
	end := len(arr) - 1

	for start < end {
		arr[start], arr[end] = arr[end], arr[start]
		start++
		end--
	}
    return arr
}

// Example usage
func reverseArrayMain() {
	arr := []int{1, 2, 3, 4, 5}
	fmt.Println("Original array:", arr)
	reversedArr := reverseArray(arr)
	fmt.Println("Reversed array:", reversedArr)
}

File: Array/Basics/ReverseArray.py
def reverse_array(arr):
    start = 0
    end = len(arr) - 1
    
    while start < end:
        arr[start], arr[end] = arr[end], arr[start]
        start += 1
        end -= 1
    return arr

# Example usage
if __name__ == "__main__":
    arr = [1, 2, 3, 4, 5]
    print("Original array:", arr)
    reversed_arr = reverse_array(arr)
    print("Reversed array:", reversed_arr)